File: mechanical/src/main/scala/resonantinduction/mechanical/energy/turbine/TurbineMBlockHandler.java
package resonantinduction.mechanical.energy.turbine;

import net.minecraft.tileentity.TileEntity;
import resonant.lib.multiblock.MultiBlockHandler;
import universalelectricity.api.vector.Vector3;

public class TurbineMBlockHandler extends MultiBlockHandler<TileTurbine>
{
    public TurbineMBlockHandler(TileTurbine wrapper)
    {
        super(wrapper);
    }

    public TileTurbine getWrapperAt(Vector3 position)
    {
        TileEntity tile = position.getTileEntity(self.getWorld());

        if (tile != null && wrapperClass.isAssignableFrom(tile.getClass()))
        {
            if (((TileTurbine) tile).getDirection() == self.getDirection() && ((TileTurbine) tile).tier == self.tier)
            {
                return (TileTurbine) tile;
            }
        }

        return null;
    }
}

File: src/main/scala/edx/mechanical/mech/turbine/TurbineMBlockHandler.scala
package edx.mechanical.mech.turbine

import net.minecraft.tileentity.TileEntity
import resonantengine.lib.prefab.tile.multiblock.reference.MultiBlockHandler
import resonantengine.lib.transform.vector.Vector3

class TurbineMBlockHandler(wrapper: TileTurbine) extends MultiBlockHandler[TileTurbine](wrapper)
{
  override def getWrapperAt(position: Vector3): TileTurbine =
  {
    val tile: TileEntity = position.getTileEntity(this.tile.getWorld)
    if (tile != null && wrapperClass.isAssignableFrom(tile.getClass))
    {
      if (tile.asInstanceOf[TileTurbine].getDirection == this.tile.getDirection && tile.asInstanceOf[TileTurbine].tier == this.tile.tier)
      {
        return tile.asInstanceOf[TileTurbine]
      }
    }
    return null
  }
}